server.py

```diff
@@ -0,0 +1,352 @@
+        >> curl -H "Content-Type: application/json" -X GET -d "{\"order_status\":\"'Checkedout'\"}" "localhost:8080/employee/sum_revenue_today"
+        """
+        order_status = flask.request.json["order_status"]
+        sum_rev_today = fos.sum_revenue_today(order_status)
+        for sum in sum_rev_today:
+            revObj = sum[0]
+            rev_dict = {
+                "today_revenue": revObj
+            }
+            return json.dumps(rev_dict)
+        print("Not found")
+        return json.dumps(not_found), 404
+
+    @app.route('/employee/delete_order', methods=['DELETE'])
+    def delete_order():
+        """
+        >> curl -H "Content-Type: application/json" -XDELETE -d "{\"order_id\":2}" "localhost:8080/employee/delete_order"
+        """
+        order_id = flask.request.json["order_id"]
+        delete = fos.delete_order(order_id)
+        return json.dumps("{}".format(success if delete else failed))
+    
+    ### customer routes
+    @app.route('/customer/view_menu', methods=['GET'])
+    def view_menu():
+        """
+        >> curl localhost:8080/customer/view_menu
+        """
+        menu = fos.view_menu()
+
+        # List of food objects to be returned to the client
+        result = []
+
+        # Loop through each menu item and convert each to a dict
+        # to be compatible for json.
+        for m in menu:
+            # m is a tuple(FoodCategory, FoodDetails)
+            # Get the FoodCategory obj from the menu
+            foodCatObj = m[0]
+            # Get the FoodDetails obj from the menu
+            foodDetObj = m[1]
+
+            # Add converted dict to resulting menu list
+            foodDictObj = {
+                "category_id": foodCatObj.category_id,
+                "category_name": foodCatObj.name,
+                "food_id": foodDetObj.food_id,
+                "food_name": foodDetObj.food_name,
+                "price": foodDetObj.price
+                }
+
+            # Add converted dict to resulting menu list
+            result.append(foodDictObj)
+        return json.dumps(result)
+
+
+    @app.route('/customer/signup', methods=['POST'])
+    def customer_signup():
+        """
+        >> curl -H "Content-Type: application/json" -X POST -d "{\"cust_name\":\"test\", \"cust_phone\":11, \"cust_email\":\"test\"}" "localhost:8080/customer/signup"
+        """
+        # Customer information saved in objects
+        cust_name = flask.request.json["cust_name"]
+        cust_phone = flask.request.json["cust_phone"]
+        cust_email = flask.request.json["cust_email"]
+
+        # Information sent to controller to be stored in database
+        cust = fos.customer_signup(cust_name, cust_phone, cust_email)
+
+        # Information returned to client
+        cust_obj = {
+            "customer_id": cust.cust_id,
+            "customer_name": cust.cust_name
+            }
+        return json.dumps(cust_obj)
+
+    @app.route('/customer/login/<int:cust_id>', methods=['GET'])
```

Comment:
In this path, if `cust_id` is the parameter for customer id, it should always be closer to the customer path. So `/customer/login/<int:cust_id>` should be `/customer/<int:cust_id>/login` instead.

Reply:
Other examples can be like:

/apples/<int:apple_id>/mangoes/<int:mango_id>
/courses/<int:course_id>/add
/courses/<int:course_id>/students/<int:student_id>/assign


Also the resources are usually plural form convetion, so `/customers`, `/employees`, `/courses`, `/students`.

Reply:
Update ALL url paths with the above two conventions. also update the sample rest api in the comments.

Reply:
updated url